de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/ESRDataLoaderUtil.java

```diff
@@ -218,23 +218,24 @@ private void setValuesFromESRString(
 
 		if (bPartner != null)
 		{
-			// check organization
-			// we should not allow matching form other org
-			if (bPartner.getAD_Org_ID() > 0 // task 09852: a partner that has no org at all does not mean an inconsistency and is therefore OK
-					&& bPartner.getAD_Org_ID() != importLine.getAD_Org_ID())
+			final boolean match = Services.get(IESRLineHandlersService.class).applyESRMatchingBPartner(bPartner, importLine);
+			
+			// check the org: should not match with invoices from other orgs
+			if (match)
+			{
+				bPartnerId = bPartner.getC_BPartner_ID();
+				importLine.setC_BPartner(bPartner);
+				importLine.setESR_Document_Status(X_ESR_ImportLine.ESR_DOCUMENT_STATUS_PartiallyMatched);
+				
+			}
+			else
```

Comment:
Hm.. you are right about the message, but in the message we need to make sure that the bp is not set.